src/core/classes/contract-events.ts

```diff
@@ -267,10 +284,15 @@ export class ContractEvents<TContract extends BaseContract> {
     event: EventFragment,
     args: ReadonlyArray<any>,
   ): ContractEvent {
-    const results: Record<string, any> = {};
+    const transaction = Object.fromEntries(
+      Object.entries(event).filter(
+        (a) => typeof a[1] !== "function" && a[0] !== "args",
```

Comment:
if i remember correctly, the reason why we have 2 of these functions is because the format is slightly different between the 2. just double check.

also what's this `function` in there?

Reply:
`event` has a bunch of FNs on it by default to parse more stuff, etc. I saw no reason to return those out with the data